File: js/1.12.js
const pascal = (i, j) => {
    if (i === j || j === 1) {
        return 1
    } else {
        return pascal(i - 1, j) + pascal(i - 1, j - 1)
    }
}

console.assert(pascal(1,1) === 1)
console.assert(pascal(3,2) === 2)

File: 1.12.clj
; write procedure that calcs i j value of pascal triangle

(defn pascal-triangle-item [i j]
  (cond (> j i) 0
        (or (= j 1) (= i j)) 1
        :else (+ (pascal-triangle-item (- i 1) j) (pascal-triangle-item (- i 1) (- j 1)))))

(assert (= (pascal-triangle-item 1 1) 1))
(assert (= (pascal-triangle-item 2 1) 1))
(assert (= (pascal-triangle-item 2 2) 1))
(assert (= (pascal-triangle-item 3 2) 2))
(assert (= (pascal-triangle-item 3 1) 1))
(assert (= (pascal-triangle-item 3 3) 1))
(assert (= (pascal-triangle-item 4 1) 1))
(assert (= (pascal-triangle-item 4 4) 1))
(assert (= (pascal-triangle-item 4 2) 3))
(assert (= (pascal-triangle-item 4 3) 3))